src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,32 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithForm extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._popupForm = this._popupEl.querySelector(".popup__form");
+    // this._handleFormSubmit = handleFormSubmit;
+    this._deleteBtn = this._popupEl.querySelector(".popup__save");
+  }
+
+  //
+  setSubmitAction(action) {
```

Comment:
Great job implementing confirmation popup and the functionality related to it! In general, it can support any confirmation.

Reply:
Thank you!